pkg/agent/controller/service_import.go

```diff
@@ -313,17 +316,31 @@ func (c *ServiceImportController) Distribute(obj runtime.Object) error {
 	result, err := util.CreateOrUpdate(context.Background(), resource.ForDynamic(c.serviceImportAggregator.brokerServiceImportClient()),
 		c.converter.toUnstructured(aggregate),
 		func(obj runtime.Object) (runtime.Object, error) {
-			// TODO - check for service type conflict.
+			existing := c.converter.toServiceImport(obj)
+
+			if localServiceImport.Spec.Type != existing.Spec.Type {
+				conflict = true
+				conflictCondition := newServiceExportCondition(
+					mcsv1a1.ServiceExportConflict, corev1.ConditionTrue, typeConflictReason,
+					fmt.Sprintf("The service type %q does not match the type (%q) of the oldest service export",
```

Comment:
fmt.Sprintf("The service type %q does not match the type (%q) of the existing service export",